lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -201,6 +201,21 @@ namespace Js
             DynamicType::New(scriptContext, TypeIds_Object, typedArrayPrototype, nullptr,
                 DeferredTypeHandler<InitializeCharArrayPrototype, DefaultDeferredTypeFilter, true>::GetDefaultInstance()));
 
+        if (scriptContext->GetConfig()->IsESSharedArrayBufferEnabled())
+        {
+            sharedArrayBufferPrototype = DynamicObject::New(recycler,
+                DynamicType::New(scriptContext, TypeIds_Object, objectPrototype, nullptr,
+                    DeferredTypeHandler<InitializeSharedArrayBufferPrototype, DefaultDeferredTypeFilter, true>::GetDefaultInstance()));
+
+            sharedArrayBufferType = DynamicType::New(scriptContext, TypeIds_SharedArrayBuffer, sharedArrayBufferPrototype, nullptr,
```

Comment:
Not completely sure about the flow here. I have copied as the exact manner on how ArrayBufferType was created. If just moving does not work out then I'll leave this as is and investigate later.

---
In reply to: [77414757](https://github.com/Microsoft/ChakraCore/pull/1533#discussion_r77414757) [](ancestors = 77414757)